### PR TITLE
Sam/com 145 move ingredients

### PR DESCRIPTION
**Pull request by asifmu97**

### CODE REVIEW

File: src/services/Backend.ts [hunk 1]
```diff
@@ -283,14 +283,71 @@ export class Backend implements BackendInterface {
...
+                        }
+                    );
+                    console.error(`Error: Received status code ${removeResponse.status}`);
+                }
+            } 
+            else{
+                console.error(`Error: Received status code ${addResponse.status}`);
+            }
+        }catch(error){
+            console.error("Failed to move ignredient:", error);
```
[the remark this answers is not in the document]
Comment (asifmu97) [reply src/services/Backend.ts:349]: fixed, thank you!

File: src/models/List.ts [hunk 2]
```diff
@@ -2,25 +2,36 @@
...
+    }
+
+    removeIngredient(ingredient: Ingredient): void {
+        const ingredientIndex = this.ingredients.findIndex(i => i.equalTo(ingredient));
+        if (ingredientIndex === -1) {
+            console.error(`Ingredient ${ingredient.name} not found in ${this.name}.`);
+            return;
+        }
+        this.ingredients.splice(ingredientIndex, 1);
+        console.log(`Removed ${ingredient.name} from ${this.name}.`);
```
[the remark this answers is not in the document]
Comment (asifmu97) [reply src/models/List.ts:34]: Got rid of most console logs (except for the Auth0 user. 

Will look into unit tests for sprint 3. Thank you for the suggestions/improvements!

File: src/components/ListNav/ListNav.tsx [hunk 3]
```diff
@@ -494,71 +531,92 @@ function ListNav({ userAuth, backendInterface }: ListNavProps) {
...
+                    >
+                        {units.map((unitOption) => ( 
+                            <MenuItem key={unitOption} value={unitOption}>
+                                {unitOption}
+                            </MenuItem>
+                        ))}
+                    </TextField>
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={handleAdd} className="primary-color" style={{ color: 'black' }}>
```
[the remark this answers is not in the document]
Comment (asifmu97) [reply src/components/ListNav/ListNav.tsx:579]: Good catch! Added and fixed.
Thank youuuu

File: src/components/ListNav/ListNav.tsx [hunk 4]
```diff
@@ -27,32 +27,39 @@ import { AiOutlineArrowLeft } from "react-icons/ai";
...
+    const [openUnitDialog, setOpenUnitDialog] = useState(false);
+    const [openConfirmDialog, setOpenConfirmDialog] = useState(false); 
+    const [ingredientToEdit, setIngredientToEdit] = useState<Ingredient | null>(null);
+    const [openEditDialog, setOpenEditDialog] = useState(false);
+    const [editAmount, setEditAmount] = useState<number | ''>('');
+    const [editUnit, setEditUnit] = useState<string>("g");
+    const [amountError, setAmountError] = useState<string>('');
+    const [searchQuery, setSearchQuery] = useState<string>("");
+    const [dialogSearchQuery, setDialogSearchQuery] = useState<string>("");
+    const [userLists, setUserLists] = useState<List[]>([]); // To store the user's lists
```
Comment (h-ng) [inline src/components/ListNav/ListNav.tsx:57]: Sorry, another real quick thing. This state variable is not used anywhere. could we remove it please? Other than that, everything else looks great!

Comment (h-ng) [inline src/components/ListNav/ListNav.tsx:57]: Sorry, another real quick thing. This state variable is not used anywhere. could we remove it please? Other than that, everything else looks great!

**Review by h-ng — CHANGES_REQUESTED** [review]

File: src/components/ListNav/ListNav.tsx (hunk 4 above)
Comment (asifmu97) [reply src/components/ListNav/ListNav.tsx]: yup sorry about that! good catch again :D
fixed

**Review by h-ng — APPROVED** [review]
Lgtm, nice work 👏💪!

**Pull request merged**